modules/sdk-coin-near/src/near.ts

```diff
@@ -1077,4 +1093,207 @@ export class Near extends BaseCoin {
+    const transactionData = transaction.toJson();
+
+    if (!transactionData.receiverId) {
+      throw new Error('Error on token enablements: missing receiver ID in transaction');
+    }
+
+    const recipient = txParams.recipients?.[0];
+    if (!recipient?.tokenName) {
+      throw new Error('Error on token enablements: missing token name in transaction parameters');
+    }
+
+    const tokenInstance = nearUtils.getTokenInstanceFromTokenName(recipient.tokenName);
+    if (!tokenInstance) {
+      throw new Error(`Error on token enablements: unknown token '${recipient.tokenName}'`);
+    }
+
+    if (transactionData.receiverId !== tokenInstance.contractAddress) {
+      throw new Error(
+        `Error on token enablements: receiver contract mismatch - expected '${tokenInstance.contractAddress}', got '${transactionData.receiverId}'`
+      );
+    }
+  }
+
+  // Validates token output information from explained transaction
+  private validateTokenOutput(explainedTx: TransactionExplanation, txParams: TransactionParams): void {
+    if (!explainedTx.outputs || explainedTx.outputs.length !== 1) {
+      throw new Error('Error on token enablements: transaction must have exactly 1 output');
+    }
+
+    const output = explainedTx.outputs[0];
+    const recipient = txParams.recipients?.[0];
+
+    if (!output.tokenName) {
+      throw new Error('Error on token enablements: missing token name in transaction output');
+    }
+
+    const tokenInstance = nearUtils.getTokenInstanceFromTokenName(output.tokenName);
+    if (!tokenInstance) {
+      throw new Error(`Error on token enablements: unknown token '${output.tokenName}'`);
+    }
+
+    if (recipient?.tokenName && recipient.tokenName !== output.tokenName) {
+      throw new Error(
+        `Error on token enablements: token mismatch - user expects '${recipient.tokenName}', transaction has '${output.tokenName}'`
+      );
+    }
+  }
+
+  private validatePublicKey(transaction: Transaction): void {
+    const transactionData = transaction.toJson();
+
+    if (!transactionData.publicKey) {
+      throw new Error('Error on token enablements: missing public key in transaction');
+    }
+
+    // Validate ed25519 format: "ed25519:base58_encoded_key"
+    if (!transactionData.publicKey.startsWith('ed25519:')) {
+      throw new Error('Error on token enablements: unsupported key type, expected ed25519');
+    }
+
+    // Validate base58 part after "ed25519:"
+    const base58Part = transactionData.publicKey.substring(8);
+    if (!base58Part || base58Part.length !== 44) {
+      // ed25519 keys are 32 bytes = 44 base58 chars
+      throw new Error('Error on token enablements: invalid ed25519 public key format');
+    }
+
+    // Validate it's actually valid base58
+    try {
+      const decoded = nearAPI.utils.serialize.base_decode(base58Part);
+      if (decoded.length !== 32) {
+        throw new Error('Error on token enablements: invalid ed25519 public key length');
+      }
```

Comment:
This throw got masked by the catch below as you throw inside the same try as the serialize.base_decode call.
Something like this could do the job!

```suggestion
      let decoded;
      try{
         decoded = nearAPI.utils.serialize.base_decode(base58Part);
      } catch {
         throw new Error('Error on token enablements: invalid base58 encoding in public key');
      }
      
      if (!decoded || decoded.length !== 32){
         throw new Error('Error on token enablements: invalid ed25519 public key length');
      }
```